### PR TITLE
backport: bitcoin#24276, #24383, #28561, #24461

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Backport

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: doc/dependencies.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ These are the dependencies currently used by Dash Core. You can find instruction
 | Dependency | Version used | Minimum required | CVEs | Shared | [Bundled Qt library](https://doc.qt.io/qt-5/configure-options.html#third-party-libraries) |
 | --- | --- | --- | --- | --- | --- |
 | Berkeley DB | [4.8.30](https://www.oracle.com/technetwork/database/database-technologies/berkeleydb/downloads/index.html) | 4.8.x | No |  |  |
-| Boost | [1.73.0](https://www.boost.org/users/download/) | [1.64.0](https://github.com/bitcoin/bitcoin/pull/22320) | No |  |  |
+| Boost | [1.77.0](https://www.boost.org/users/download/) | [1.64.0](https://github.com/bitcoin/bitcoin/pull/22320) | No |  |  |
```
Comment (knst) [inline doc/dependencies.md:9]:
```suggestion
| Boost | [1.77.0](https://www.boost.org/users/download/) | [1.77.0](https://github.com/bitcoin/bitcoin/pull/24383) | No |  |  |
```

Comment (PastaPastaPasta) [reply doc/dependencies.md:9]: can you clarify what you mean?

File: depends/packages/boost.mk [hunk 2]
```diff
@@ -1,8 +1,8 @@
 package=boost
-$(package)_version=1_73_0
-$(package)_download_path=https://boostorg.jfrog.io/artifactory/main/release/$(subst _,.,$($(package)_version))/source/
-$(package)_file_name=boost_$($(package)_version).tar.bz2
-$(package)_sha256_hash=4eb3b8d442b426dc35346235c8733b5ae35ba431690e38c6a8263dce9fcbb402
+$(package)_version=1.77.0
```
Comment (UdjinM6) [inline depends/packages/boost.mk:2]: package version format changes should be done via 24276 backport and not squashed silently into 24383

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
mac build is broken, probably because of 28337

it's also weird that no all GH checks are running here

**Comment by UdjinM6** [comment]
28561 missing changes in `doc/dependencies.md`

**Review by knst — APPROVED** [review]
utACK

Should #5671 be closed as a duplicate ?

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**